File: ama/AstroWeather2/app/src/main/java/com/android/blawniczak/astroweather/fragments/DayWeatherFragment.java
package com.android.blawniczak.astroweather.fragments;

import android.os.Bundle;
import android.support.v7.app.AppCompatDialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.android.blawniczak.astroweather.R;
import com.android.blawniczak.astroweather.domain.weather.Forecast;

import butterknife.BindView;
import butterknife.ButterKnife;

public class DayWeatherFragment extends AppCompatDialogFragment {

    @BindView(R.id.date) TextView date;
    @BindView(R.id.day) TextView day;
    @BindView(R.id.high) TextView high;
    @BindView(R.id.low) TextView low;
    @BindView(R.id.text) TextView text;

    public static DayWeatherFragment newInstance() {
        return new DayWeatherFragment();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) { super.onCreate(savedInstanceState); }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup parent,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_day_weather, parent, false);
        ButterKnife.bind(this, v);
        return v;
    }

    public void calculateValues(Forecast forecast) {
        if(this.date != null) {
            this.date.setText(forecast.date);
            this.day.setText(forecast.day);
            this.high.setText(forecast.high);
            this.low.setText(forecast.low);
            this.text.setText(forecast.text);
        }
    }
}

File: paiwjs/AstroWeather2/app/src/main/java/com/android/blawniczak/astroweather/fragments/DayWeatherFragment.kt
package com.android.blawniczak.astroweather.fragments

import android.os.Bundle
import android.support.v7.app.AppCompatDialogFragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.android.blawniczak.astroweather.R
import com.android.blawniczak.astroweather.domain.weather.Forecast

import butterknife.BindView
import butterknife.ButterKnife

class DayWeatherFragment : AppCompatDialogFragment() {

    @BindView(R.id.date)
    @JvmField internal var date: TextView? = null
    @BindView(R.id.day)
    @JvmField internal var day: TextView? = null
    @BindView(R.id.high)
    @JvmField internal var high: TextView? = null
    @BindView(R.id.low)
    @JvmField internal var low: TextView? = null
    @BindView(R.id.text)
    @JvmField internal var text: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, parent: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.fragment_day_weather, parent, false)
        ButterKnife.bind(this, v)
        return v
    }

    fun calculateValues(forecast: Forecast) {
        if (this.date != null) {
            this.date!!.text = forecast.date
            this.day!!.text = forecast.day
            this.high!!.text = forecast.high
            this.low!!.text = forecast.low
            this.text!!.text = forecast.text
        }
    }

    companion object {

        fun newInstance(): DayWeatherFragment {
            return DayWeatherFragment()
        }
    }
}
